examples/crm/src/companies/CompanyActivityIterator.tsx

```diff
@@ -0,0 +1,330 @@
+
+export function CompanyActivityIterator() {
+    const company = useRecordContext<Company>();
+    const { data: activity, error, isPending } = useListContext<Activity>();
+    if (!company || isPending || error) return null;
+
+    return (
+        <Box>
+            <List
+                sx={{
+                    '& .MuiListItem-root': {
+                        justifyContent: 'space-between',
+                        gap: 2,
+                    },
+                    '& .MuiListItem-root:not(:first-child)': {
+                        borderTop: '1px solid #f0f0f0',
+                    },
+                }}
+            >
+                {activity.map(activity => (
+                    <ActivityItem
+                        key={activity.id}
+                        activity={activity}
+                        company={company}
+                    />
+                ))}
+            </List>
+        </Box>
+    );
+}
+
+function ActivityItem({
+    activity,
+    company,
+}: {
+    activity: Activity;
+    company: Company;
+}) {
+    switch (activity.type) {
+        case COMPANY_CREATED:
+            return (
+                <CompanyCreatedActivity activity={activity} company={company} />
+            );
+        case CONTACT_CREATED:
+            return (
+                <ContactCreatedActivity activity={activity} company={company} />
+            );
+        case DEAL_CREATED:
+            return (
+                <DealCreatedActivity activity={activity} company={company} />
+            );
+        case CONTACT_NOTE_CREATED:
+            return (
+                <ContactNoteCreatedActivity
+                    activity={activity}
+                    company={company}
+                />
+            );
+        default:
+            return null;
+    }
+}
+
+function DisplayDate({ date }: { date: string }) {
+    const now = new Date();
+    const d = new Date(date);
+
+    if (
+        d.getDay() === now.getDay() &&
+        d.getMonth() === now.getMonth() &&
+        d.getFullYear() === now.getFullYear()
+    ) {
+        return (
+            <Typography
+                color="textSecondary"
+                sx={{
+                    minWidth: '96px',
+                    display: 'flex',
+                    justifyContent: 'flex-end',
+                }}
+            >
+                {new Intl.DateTimeFormat('en-US', {
+                    timeStyle: 'short',
+                }).format(d)}
+            </Typography>
+        );
+    }
+
+    return (
+        <Typography
+            color="textSecondary"
+            sx={{
+                minWidth: '96px',
+                display: 'flex',
+                justifyContent: 'flex-end',
+            }}
+        >
+            {new Intl.DateTimeFormat('en-US', {
+                dateStyle: 'short',
+            }).format(d)}
+        </Typography>
+    );
+}
+
+function CompanyCreatedActivity({
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** ‏

Split into multiple file ?